packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -243,6 +264,15 @@ export default createRule<Options, MessageIds>({
         .join('\n');
 
       if (lastCase) {
+        const isLastCaseDefaultCase = lastCase.test == null;
+        if (isLastCaseDefaultCase) {
```

Comment:
(optional)

food for thought - what if the default case isn't the last case?

This change makes it so that we avoid suggesting to fix
```ts
declare const literal: 'a' | 'b';

switch (literal) {
  case 'b':
  default:
}
```
to 
```ts
declare const literal: 'a' | 'b';

switch (literal) {
  case 'b':
  default:
  case "a": { throw new Error('Not implemented yet: "a" case') }
}
```

and instead fix it to 
```ts
declare const literal: 'a' | 'b';

switch (literal) {
  case 'b':
  case "a": { throw new Error('Not implemented yet: "a" case') }
  default:
}
```
. (also note that this has its own issue of no longer having `case 'b'` be handled the same as `default`, instead having it throw as not implemented yet).

But it will happily fix 
```ts
declare const literal: 'a' | 'b';

switch (literal) {
  default:
  case 'b':
}
```
to
```ts
declare const literal: 'a' | 'b';

switch (literal) {
  default:
  case 'b':
  case "a": { throw new Error('Not implemented yet: "a" case') }
}
```
which still has the added case as part of the default 🤔, even though the first examples implies this code is trying to avoid it. Maybe to be more consistent we could say "if default case is present, add it above it, else add it at the end"?

***

I don't think that this is super important since the user will have to do manual cleanup regardless, but just throwing this out there in case you want to tweak some edge cases. I'm ok with either way though.